proxy_agent_setup/src/linux/azure-proxy-agent.service

```diff
@@ -1,5 +1,7 @@
 [Unit]
 Description=Microsoft Azure GuestProxyAgent
+# azure-proxy-agent supposed to be an early boot and late system shutdown service
+DefaultDependencies=no
```

Comment:
If you add this, I think you'll also want `Conflicts=shutdown.target` and `Before=shutdown.target` to ensure the service is properly stopped on shutdown or reboot.

And I think you will want `After=sysinit.target` to ensure that basic system initialization has completed, but it's worth a careful read of [systemd.special(7)](https://manpages.debian.org/trixie/systemd/systemd.special.7.en.html) and [bootup(7)](https://manpages.debian.org/trixie/systemd/bootup.7.en.html) to be sure.